### ISSUE TITLE
TimeDelta does not support np.sum anymore

### ISSUE THREAD

**Issue by adonath** [astropy/astropy#10501]
### Description
In the development version of Astropy the `TimeDelta` object does not work with `np.sum()` anymore. The issue was found in one of our [Gammapy Travis-CI builds](https://travis-ci.org/github/gammapy/gammapy/jobs/699226524#L8080). I presume the issue is somehow related to https://github.com/astropy/astropy/issues/8610, but I could not find a statement whether it is intended behaviour.

### Expected behavior

I would expect that the expression `np.sum(time_delta)` just returns the sum of the time differences

### Actual behavior

The code fails with a `TypeError`:

```
TypeError: no implementation found for 'numpy.sum' on types that implement __array_function__: [<class 'astropy.time.core.TimeDelta'>]
```

### Steps to Reproduce
The following code example reproduces the error:

```python
from astropy.time import Time
import numpy as np

t_1 = Time(['1999-01-01T00:00:00.123456789', '2010-01-01T00:00:00'])
t_2 = Time(['1998-01-01T00:00:00.123456789', '2009-01-01T00:00:00'])
total = np.sum(t_1 - t_2)
```

### System Details

Darwin-18.7.0-x86_64-i386-64bit
Python 3.7.0 | packaged by conda-forge | (default, Nov 12 2018, 12:34:36) 
[Clang 4.0.1 (tags/RELEASE_401/final)]
Numpy 1.18.1
astropy 4.2.dev267+g295c2ab4b
Scipy 1.4.1
Matplotlib 3.1.2

**Comment by adonath** [comment]
The same issue appears with `np.diff` and `Time` objects:
```python
from astropy.time import Time
import numpy as np

t = Time(['1999-01-01T00:00:00.123456789', '2010-01-01T00:00:00'])
np.diff(t)
```
Where I would expect the call to `np.diff()` to return a `TimeDelta` object with the difference of the elements of `t`.

**Comment by adonath** [comment]
Possible work-arounds are:
```python
from astropy import units as u
from astropy.time import Time
import numpy as np

t_1 = Time(['1999-01-01T00:00:00.123456789', '2010-01-01T00:00:00'])
t_2 = Time(['1998-01-01T00:00:00.123456789', '2009-01-01T00:00:00'])

diff = t_1 - t_2
total = u.Quantity(np.sum(diff.to_value("s")), "s", copy=False)
```
and

```python
from astropy.time import Time
import numpy as np

t = Time(['1999-01-01T00:00:00.123456789', '2010-01-01T00:00:00'])
t[1:] - t[:-1]
``` 
However this is a much less intuitive solution than the previous seamless support of `np.sum()` and `np.diff()`.

**Comment by pllim** [comment]
The "anymore" implies that this used to work? Which `astropy` version where this worked for you? Thanks for reporting this.

**Comment by adonath** [comment]
@pllim I just checked and it still works in `v4.0.1`.

**Comment by pllim** [comment]
Thanks!

**Comment by mhvk** [comment]
@adonath - yes, that is definitely related to #8610 - in #10337 I added an override for all numpy functions and any that are not on a white-list will be refused.  I think the easiest may be to continue to send things on instead for functions that are not explicitly listed.

That said, ideally we would provide better implementations of those functions, since, as is, they are extremely slow since they cast times as object arrays:
```
In [5]: t = Time(50000, np.arange(10), format='mjd')                                                   

In [6]: td = np.diff(t)                                                                                

In [7]: td                                                                                             
Out[7]: 
array([<TimeDelta object: scale='tai' format='jd' value=1.0>,
       <TimeDelta object: scale='tai' format='jd' value=1.0>,
       <TimeDelta object: scale='tai' format='jd' value=1.0>,
       <TimeDelta object: scale='tai' format='jd' value=1.0>,
       <TimeDelta object: scale='tai' format='jd' value=1.0>,
       <TimeDelta object: scale='tai' format='jd' value=1.0>,
       <TimeDelta object: scale='tai' format='jd' value=1.0>,
       <TimeDelta object: scale='tai' format='jd' value=1.0>,
       <TimeDelta object: scale='tai' format='jd' value=1.0>],
      dtype=object)

In [8]: td.sum()                                                                                       
Out[8]: <TimeDelta object: scale='tai' format='jd' value=9.0>

In [9]: td.reshape(3,3).sum(1)                                                                         
Out[9]: 
array([<TimeDelta object: scale='tai' format='jd' value=3.0>,
       <TimeDelta object: scale='tai' format='jd' value=3.0>,
       <TimeDelta object: scale='tai' format='jd' value=3.0>],
      dtype=object)
```

**Comment by mhvk** [comment]
@adonath - a fix is in #10502. It does just fix the regression, not introduce proper implementations. It might be useful to add to #8610 which functions one would expect to work.
